backend/dataall/modules/shares_base/api/types.py

```diff
@@ -0,0 +1,257 @@
+    ],
+)
+
+ShareObject = gql.ObjectType(
+    name='ShareObject',
+    fields=[
+        gql.Field(name='shareUri', type=gql.ID),
+        gql.Field(name='status', type=gql.Ref('ShareObjectStatus')),
+        gql.Field(name='owner', type=gql.String),
+        gql.Field(name='created', type=gql.String),
+        gql.Field(name='deleted', type=gql.String),
+        gql.Field(name='updated', type=gql.String),
+        gql.Field(name='datasetUri', type=gql.String),
+        gql.Field(name='requestPurpose', type=gql.String),
+        gql.Field(name='rejectPurpose', type=gql.String),
+        gql.Field(name='dataset', type=DatasetLink, resolver=resolve_dataset),
+        gql.Field(name='alreadyExisted', type=gql.Boolean),
+        gql.Field(name='existingSharedItems', type=gql.Boolean, resolver=resolve_existing_shared_items),
+        gql.Field(
+            name='statistics',
+            type=gql.Ref('ShareObjectStatistic'),
+            resolver=resolve_share_object_statistics,
+        ),
+        gql.Field(name='principal', resolver=resolve_principal, type=gql.Ref('Principal')),
+        gql.Field(
+            name='environment',
+            resolver=resolve_environment,
+            type=gql.Ref('Environment'),
+        ),
+        gql.Field(
+            name='group',
+            resolver=resolve_group,
+            type=gql.String,
+        ),
+        gql.Field(
+            'items',
+            args=[gql.Argument(name='filter', type=gql.Ref('ShareableObjectFilter'))],
+            type=gql.Ref('SharedItemSearchResult'),
+            resolver=list_shareable_objects,
+        ),
+        gql.Field(
+            name='canViewLogs',
+            resolver=resolve_can_view_logs,
+            type=gql.Boolean,
+        ),
+        gql.Field(
+            name='userRoleForShareObject',
+            type=gql.Ref('ShareObjectPermission'),
+            resolver=resolve_user_role,
+        ),
+    ],
+)
+
+
+ShareSearchResult = gql.ObjectType(
+    name='ShareSearchResult',
+    fields=[
+        gql.Field(name='count', type=gql.Integer),
+        gql.Field(name='pageSize', type=gql.Integer),
+        gql.Field(name='nextPage', type=gql.Integer),
+        gql.Field(name='pages', type=gql.Integer),
+        gql.Field(name='page', type=gql.Integer),
+        gql.Field(name='previousPage', type=gql.Integer),
+        gql.Field(name='hasNext', type=gql.Boolean),
+        gql.Field(name='hasPrevious', type=gql.Boolean),
+        gql.Field(name='nodes', type=gql.ArrayType(gql.Ref('ShareObject'))),
+    ],
+)
+
+EnvironmentPublishedItem = gql.ObjectType(
+    name='EnvironmentPublishedItem',
+    fields=[
+        gql.Field(name='shareUri', type=gql.NonNullableType(gql.String)),
+        gql.Field(name='datasetUri', type=gql.NonNullableType(gql.String)),
+        gql.Field(name='datasetName', type=gql.NonNullableType(gql.String)),
+        gql.Field(name='itemAccess', type=gql.NonNullableType(gql.String)),
```

Comment:
should itemAccess be a valid Field anymore?

Reply:
Not anymore! good catch